src/dsp_tools/fast_xmlupload/process_files.py

```diff
@@ -102,7 +103,28 @@ def _process_files_in_parallel(
 
     orig_filepath_2_uuid: list[tuple[Path, Optional[Path]]] = []
     for processed in as_completed(processing_jobs):
-        orig_filepath_2_uuid.append(processed.result())
+        try:
+            orig_filepath_2_uuid.append(processed.result())
+        except docker.errors.APIError:
+            print(f"{datetime.now()}: ERROR: A Docker exception occurred. Restart SIPI and resume processing...")
+            logger.error("A Docker exception occurred. Restart SIPI and resume processing...", exc_info=True)
+            # cancel all running jobs
+            for job in processing_jobs:
+                job.cancel()
+            # restart sipi container
+            _stop_and_remove_sipi_container()
+            _start_sipi_container_and_mount_volumes(input_dir, output_dir)
+            global sipi_container
+            sipi_container = _get_sipi_container()
+            # restart processing
+            processed_paths = [x[0] for x in orig_filepath_2_uuid]
+            unprocessed_paths = [x for x in paths if x not in processed_paths]
+            orig_filepath_2_uuid += _process_files_in_parallel(
+                paths=unprocessed_paths,
+                input_dir=input_dir,
+                output_dir=output_dir,
+                nthreads=nthreads
+            )
```

Comment:
doing a recursive call like this can potentially be a cause for errors, if the max recursion depth gets reached. In general, it would probably be more robust to move calling this function again one "level out", so it's flat, not recursive.  
But, to be honest, I'm not sure how big the danger is of this actually happening

Reply:
To be safe, I moved calling this function again one level out. Can you check that there's no logical flaw in it?

Reply:
I didn't look thoroughly, but at first glance it seems good